### PR TITLE
Remote JMX Monitor

### PR DESCRIPTION
**Pull request by diegolovison**

### CODE REVIEW

File: docs/stages/core.md [hunk 1]
```diff
@@ -240,6 +240,19 @@ Periodically generates heap dumps.
 > stop (**optional**) - Set this flag to true in order to terminate the heap dumper. Default is false.  
 > workers (**optional**) - Specifies on which workers this stage should actively run. The result set is intersection of specified workers, groups and roles. Default is all workers.  
 
+### remote-monitor-start
+Starts collecting statistics locally on main and each worker node.
+> exit-on-failure (**optional**) - If true, then the benchmark stops when the stage returns an error. If false, then the stages in the current scenario are skipped, and the next scenario starts executing. Default is false.  
+> groups (**optional**) - Specifies in which groups this stage should actively run. The result set is intersection of specified workers, groups and roles. Default is all groups.  
+> interface-name (**optional**) - Specifies the network interface where statistics are gathered. If not specified, then statistics are not collected.  
+> jmx-service-url (**optional**) - Remove JMX service url. Example: service:jmx:rmi:///jndi/rmi://127.0.0.1:9999/jmxrmi
```
Comment (alanfx) [inline docs/stages/core.md:248]: Remove->Remote

File: core/src/main/java/org/radargun/stages/monitor/RemoteMonitorStartStage.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.radargun.config.Stage;
+import org.radargun.traits.InternalsExposition;
+import org.radargun.traits.JmxConnectionProvider;
+
+@Stage(doc = "Starts collecting statistics locally on main and each worker node.", deprecatedName = "jvm-monitor-start")
+public class RemoteMonitorStartStage extends AbstractMonitorStartStage {
+
+   private JmxConnectionProvider jmxConnectionProvider;
+
+   @Property(doc = "Remove JMX service url. Example: service:jmx:rmi:///jndi/rmi://127.0.0.1:9999/jmxrmi")
```
Comment (alanfx) [inline core/src/main/java/org/radargun/stages/monitor/RemoteMonitorStartStage.java:20]: Remove->Remote

File: core/src/main/java/org/radargun/stages/monitor/AbstractMonitorStartStage.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import org.radargun.traits.JmxConnectionProvider;
+import org.radargun.utils.SystemUtils;
+import org.radargun.utils.TimeConverter;
+
+/**
+ *
+ * Starts collecting JVM statistics locally on main and each worker node.
+ * {@link SystemWorkerMonitor}
+ *
+ * @author Alan Field &lt;afield@redhat.com&gt;
```
Comment (alanfx) [inline core/src/main/java/org/radargun/stages/monitor/AbstractMonitorStartStage.java:27]: Not sure I should be the `@author` here!

**Review by alanfx — CHANGES_REQUESTED** [review]
Hey @diegolovison looks good. Just a couple of small things, and then I can merge

File: core/src/main/java/org/radargun/stages/monitor/AbstractMonitorStartStage.java (hunk 3 above)
Comment (diegolovison) [reply core/src/main/java/org/radargun/stages/monitor/AbstractMonitorStartStage.java:27]: The `AbstractMonitorStartStage` is basically `MonitorStartStage` that has your name. I decided to keep it :)

File: docs/stages/core.md (hunk 1 above)
Comment (diegolovison) [reply docs/stages/core.md]: Done

File: core/src/main/java/org/radargun/stages/monitor/RemoteMonitorStartStage.java (hunk 2 above)
Comment (diegolovison) [reply core/src/main/java/org/radargun/stages/monitor/RemoteMonitorStartStage.java]: Done

**Review by gustavolira — APPROVED** [review]

**Comment by gustavolira** [comment]
#722 merged into Master, thanks Diego!

**Pull request merged**